src/main/java/kg/jarkyn/server/ReceivingSocket.java

```diff
@@ -0,0 +1,22 @@
+package kg.jarkyn.server;
+
+import kg.jarkyn.server.Incoming.Requester;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+
+public class ReceivingSocket {
+    private ServerSocket serverSocket;
+
+    public ReceivingSocket(ServerSocket serverSocket) {
```

Comment:
i'll have a look at what you are suggesting, so far was making sure that wrappers are very thin and are just delegating. Initially my Receiving socket was returning client's input stream, rather than real client itself, but here's why I changed it:

1) any logic of my own would require tests and there's this catch 22 again, because I would then need to use the real socket in tests

2) trying to avoid extra mental mapping especially for those familiar with the api of the sockets